src/nvim/eval/funcs.c

```diff
@@ -8053,14 +8055,16 @@ static void f_screenchar(typval_T *argvars, typval_T *rettv, FunPtr fptr)
 {
   int c;
 
+  ScreenGrid *grid;
   int row = tv_get_number_chk(&argvars[0], NULL) - 1;
   int col = tv_get_number_chk(&argvars[1], NULL) - 1;
-  if (row < 0 || row >= default_grid.rows
-      || col < 0 || col >= default_grid.cols) {
+
+  screenchar_adjust(&grid, &row, &col);
+
+  if (row < 0 || row >= grid->rows
+      || col < 0 || col >= grid->cols) {
```

Comment:
nit: can be on 1 line (same for the other case)

Reply:
Done.